exporters/otlp-http/trace/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporterBuilder.java

```diff
@@ -0,0 +1,161 @@
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import okhttp3.Headers;
+import okhttp3.OkHttpClient;
+import okhttp3.tls.HandshakeCertificates;
+
+/** Builder utility for this exporter. */
+public final class OtlpHttpSpanExporterBuilder {
+
+  private static final long DEFAULT_TIMEOUT_SECS = 10;
+  private static final String DEFAULT_ENDPOINT = "http://localhost:4317/v1/traces";
+
+  private long timeoutNanos = TimeUnit.SECONDS.toNanos(DEFAULT_TIMEOUT_SECS);
+  private String endpoint = DEFAULT_ENDPOINT;
+  private boolean isCompressionEnabled = false;
+  @Nullable private Headers.Builder headersBuilder;
+  @Nullable private byte[] trustedCertificatesPem;
+
+  /**
+   * Sets the maximum time to wait for the collector to process an exported batch of spans. If
+   * unset, defaults to {@value DEFAULT_TIMEOUT_SECS}s.
+   */
+  public OtlpHttpSpanExporterBuilder setTimeout(long timeout, TimeUnit unit) {
+    requireNonNull(unit, "unit");
+    checkArgument(timeout >= 0, "timeout must be non-negative");
+    timeoutNanos = unit.toNanos(timeout);
+    return this;
+  }
+
+  /**
+   * Sets the maximum time to wait for the collector to process an exported batch of spans. If
+   * unset, defaults to {@value DEFAULT_TIMEOUT_SECS}s.
+   */
+  public OtlpHttpSpanExporterBuilder setTimeout(Duration timeout) {
+    requireNonNull(timeout, "timeout");
+    return setTimeout(timeout.toNanos(), TimeUnit.NANOSECONDS);
+  }
+
+  /**
+   * Sets the OTLP endpoint to connect to. If unset, defaults to {@value DEFAULT_ENDPOINT}. The
+   * endpoint must start with either http:// or https://, and should append the version and signal
+   * to the path (i.e. v1/traces).
+   */
+  public OtlpHttpSpanExporterBuilder setEndpoint(String endpoint) {
+    requireNonNull(endpoint, "endpoint");
+
+    URI uri;
+    try {
+      uri = new URI(endpoint);
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException("Invalid endpoint, must be a URL: " + endpoint, e);
+    }
+
+    if (uri.getScheme() == null
+        || (!uri.getScheme().equals("http") && !uri.getScheme().equals("https"))) {
+      throw new IllegalArgumentException(
+          "Invalid endpoint, must start with http:// or https://: " + uri);
+    }
+
+    this.endpoint = endpoint;
+    return this;
+  }
+
+  /**
+   * Sets the method used to compress payloads. If unset, compression is disabled. Currently the
+   * only supported compression method is "gzip".
+   */
+  public OtlpHttpSpanExporterBuilder setCompression(String compressionMethod) {
+    requireNonNull(compressionMethod, "compressionMethod");
+    Preconditions.checkArgument(
+        "gzip".equals(compressionMethod),
```

Comment:
We prefer to use yoda order only when needed. We checked null so the variable can be on left side.

Reply:
I personally prefer "yoda" order. Have we codified this style choice anywhere?

Reply:
Don't think so, we probably could. I would have probably commented if you did it somewhere :P `if` statements read terribly with yoda order IMO